aes.js

```diff
@@ -81,14 +81,17 @@ AES.keySize = 256 / 8
 AES.prototype.keySize = AES.keySize
 
 function bufferToArray (buf) {
-  var len = buf.length / 4
+  if (!Buffer.isBuffer(buf)) {
```

Comment:
I'm pretty sure bufferToArray is an internal function so it may make more sense to put the check outside the function because we may be calling it in situations where we are confident that it is a buffer